### PR TITLE
Fix dockerstat network metrics

### PR DESCRIPTION
**Pull request by ChristianKniep**
I changed the dockerstat network collector to pick up the per-network-interface metrics instead of the overall metrics, which are 0 in the newest version of the docker api.
I ran a small iperf benchmark between two containers to see if it reports correctly - LGTM.

```
[root@grafana /]# iperf -c carbon -t 300
------------------------------------------------------------
Client connecting to carbon, TCP port 5001
TCP window size: 85.0 KByte (default)
------------------------------------------------------------
[  3] local 172.17.0.5 port 42773 connected with 172.17.0.4 port 5001
[ ID] Interval       Transfer     Bandwidth
[  3]  0.0-300.0 sec   814 GBytes  23.3 Gbits/sec
[root@grafana /]#
```
screen shot 2016-03-04 at 17 46 14

### CODE REVIEW

File: src/fullerite/collector/docker_stats.go [hunk 1]
```diff
@@ -146,13 +146,17 @@ func (d DockerStats) getDockerContainerInfo(container *docker.Container) {
 
 // buildMetrics creates the actual metrics for the given container.
 func (d DockerStats) buildMetrics(container *docker.Container, containerStats *docker.Stats, cpuPercentage float64) []metric.Metric {
+	//var netiface string
```
Comment (tsheasha) [inline src/fullerite/collector/docker_stats.go]: debug?

File: src/fullerite/collector/docker_stats.go [hunk 2]
```diff
@@ -146,13 +146,17 @@ func (d DockerStats) getDockerContainerInfo(container *docker.Container) {
...
 	ret := []metric.Metric{
-		buildDockerMetric("DockerRxBytes", metric.CumulativeCounter, float64(containerStats.Network.RxBytes)),
-		buildDockerMetric("DockerTxBytes", metric.CumulativeCounter, float64(containerStats.Network.TxBytes)),
 		buildDockerMetric("DockerMemoryUsed", metric.Gauge, float64(containerStats.MemoryStats.Usage)),
 		buildDockerMetric("DockerMemoryLimit", metric.Gauge, float64(containerStats.MemoryStats.Limit)),
+		buildDockerMetric("DockerCpuUser", metric.Gauge, cpuPercentage),
```
Comment (tsheasha) [inline src/fullerite/collector/docker_stats.go]: this is the same as "DockerCpuPercentage" on the next line, for the sake of not breaking so many time-series and dashboards users already have under this name, I say we keep it as it is. What do you think?

File: src/fullerite/collector/docker_stats.go [hunk 3]
```diff
@@ -146,13 +146,17 @@ func (d DockerStats) getDockerContainerInfo(container *docker.Container) {
...
 		buildDockerMetric("DockerCpuPercentage", metric.Gauge, cpuPercentage),
 	}
+	for netiface, _ := range containerStats.Networks {
+		ret = append(ret, buildDockerMetric(fmt.Sprintf("DockerTxBytes.%s", netiface), metric.CumulativeCounter, float64(containerStats.Networks[netiface].TxBytes)))
```
Comment (tsheasha) [inline src/fullerite/collector/docker_stats.go]: The `netiface` is more a dimension that it is a part of the metric name. See for example:
https://github.com/Yelp/fullerite/blob/master/src/diamond/collectors/network/network.py#L126-L128
Same applies for the RxBytes metric
Makes sense?

**Comment by tsheasha** [comment]
Other than the comments I added, looks good. Please also fix the tests for that collector to make the build succeed.

File: src/fullerite/collector/docker_stats.go (hunk 2 above)
Comment (poros) [inline src/fullerite/collector/docker_stats.go]: I agree with @tsheasha.

File: src/fullerite/collector/docker_stats.go (hunk 3 above)
Comment (avadhutp) [inline src/fullerite/collector/docker_stats.go]: Seconded. Please add `netiface` as a dimension to the metric.

**Comment by drolando** [comment]
Have you tested this change with an older version of docker? AFAIK network stats per interface were introduced in docker 1.9.x.

We still have some hosts with docker 1.6, so this change would probably break fullerite on them..

**Comment by tsheasha** [comment]
@drolando the docker lient library we use handles that no?

**Comment by drolando** [comment]
I just noticed that they fixed the go-dockerclient library and made it backward compatible.

However we'll still need to test it with an older docker version to verify that it doesn't break and that it still emits the correct metrics.

**Comment by ChristianKniep** [comment]
Just got from the plan to hop back over the pond. Coded the test, will check and adjust the PR tomorrow.

**Comment by ChristianKniep** [comment]
Yeah! My first test succeeds... :)
```
➜  fullerite git:(fix_dockerstat_network_metrics) ✗ make test
Getting dependencies...
...............
----------------------------------------------------------------------
Ran 15 tests in 0.007s

OK
Testing fullerite
ok  	beatit	0.012s	coverage: 22.2% of statements
ok  	fullerite	3.022s	coverage: 39.4% of statements
ok  	fullerite/collector	6.149s	coverage: 80.8% of statements
ok  	fullerite/config	0.013s	coverage: 78.1% of statements
ok  	fullerite/handler	3.023s	coverage: 57.3% of statements
ok  	fullerite/internalserver	0.117s	coverage: 88.1% of statements
ok  	fullerite/metric	0.011s	coverage: 81.0% of statements
ok  	fullerite/util	0.012s	coverage: 93.2% of statements
➜  fullerite git:(fix_dockerstat_network_metrics) ✗
```

**Comment by gnufied** [comment]
I need to dig bit more but while this works on Docker version 1.10.x but it doesn't work on 1.6.0x version. It doesn't even throw error.

**Comment by drolando** [comment]
@gnufied any update on this? We have upgraded to docker 1.10 so this should be good to go

**Comment by gnufied** [comment]
I will get this merged after fixing the conflict etc. thanks @drolando !

**Comment by gnufied** [comment]
Closing this, since I have merged this branch after fixing conflicts.

**Pull request closed without merging**